idseq_dag/steps/run_alignment_remotely.py

```diff
@@ -334,211 +317,80 @@ def run_chunk_wrapper(chunks_in_flight, chunk_output_files, n, mutex, target, ar
-                    finally:
-                        # None chunk_status indicates code bug above.  An exception has been raised already
-                        # for it, and it says nothing about whether the alignment succeeded or not.
-                        if chunk_status != None:
-                            chunk_status_tracker(service).note_outcome(instance_ip, chunk_id, elapsed, chunk_status, try_number)
-                        self.__delete_remote_dir(remote_work_dir, key_path, remote_username, instance_ip)
-
-            # Upload to s3
-            with self.iostream_upload:  # Limit concurrent uploads so as not to stall the pipeline.
-                command.execute(
-                    command_patterns.SingleCommand(
-                        cmd="aws",
-                        args=[
-                            "s3",
-                            "cp",
-                            "--only-show-errors",
-                            multihit_local_outfile,
-                            os.path.join(self.chunks_result_dir_s3, "")
-                        ]
-                    )
-                )
-            log.write(f"finished alignment for chunk {chunk_id} on {service} server {instance_ip}")
+            return multihit_local_outfile
+
+        # TODO: (tmorse) remove compat hack
+        deployment_environment = os.environ.get("DEPLOYMENT_ENVIRONMENT", self.additional_attributes.get("environment"))
+        priority_name = os.environ.get("PRIORITY_NAME", "normal")
+        provisioning_model = os.environ.get("PROVISIONING_MODEL", "EC2")
+
+        index_dir_suffix = self.additional_attributes["index_dir_suffix"]
+
+        job_name = f"idseq-{deployment_environment}-{service}"
+        job_queue = f"idseq-{deployment_environment}-{service}-{provisioning_model}-{index_dir_suffix}-{priority_name}"
+        job_definition = f"idseq-{deployment_environment}-{service}"
+
+        environment = [{
+            'name': f"INPUT_PATH_{i}",
+            'value': os.path.join(self.chunks_result_dir_s3, input_file),
+        } for i, input_file in enumerate(input_files)] + [{
+            'name': "OUTPUT_PATH",
+            'value': multihit_s3_outfile,
+        }]
+
+        session = boto3.session.Session()
+        client = session.client("batch")
+        response = client.submit_job(
+            jobName=job_name,
+            jobQueue=job_queue,
+            jobDefinition=job_definition,
+            containerOverrides={"environment": environment},
+            retryStrategy={"attempts": CHUNK_MAX_ATTEMPTS},
+            timeout={"attemptDurationSeconds": CHUNK_ATTEMPT_TIMEOUT}
+        )
+        job_id = response["jobId"]
+
+        log.write(f"waiting for {service} batch queue for chunk {chunk_id}.")
+        total_timeout = CHUNK_MAX_ATTEMPTS * CHUNK_ATTEMPT_TIMEOUT
+        for _ in range((total_timeout // CHUNK_COMPLETE_CHECK_DELAY) + 1):
+            status = PipelineStepRunAlignmentRemotely._get_job_status(client, job_id)
```

Comment:
Good point. Usually there are many fewer than 32 chunks though so I don't want to create unnecessary waiting by relaxing the time too much. Do you think it would be a good idea to keep calls per second constant by factoring in the chunk count?

Reply:
Sure, as long as the number of parallel samples does not drive us close some AWD quota or throttling. 
Does boto raise an exception for throttling errors?

Reply:
Yes I believe it does

Reply:
Then we need to handle that exception, right?